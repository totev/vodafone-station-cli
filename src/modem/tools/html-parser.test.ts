import * as fs from 'fs'
import * as path from 'path'
import {
  extractCredentialString,
  extractCryptoVars,
  extractDocsisStatus,
  extractFirmwareVersion,
} from './html-parser'

describe('htmlParser', () => {
  const fixtureIndex = fs.readFileSync(
    path.join(__dirname, './__fixtures__/index.php.html'),
    'utf8'
  )
  test('extractCryptoVars', () => {
    const expected = {
      nonce: 'cyCCZrSU0MXlXGhDso44BGA+MmA=',
      iv: 'da571578b4f51e21',
      salt: '02355f4a986c6900',
      sessionId: '343fe70eb4a25c54c34ce1c43d8359f4',
    }
    const extractedVars = extractCryptoVars(fixtureIndex)
    expect(extractedVars).toBeTruthy()
    expect(extractedVars).toEqual(expected)
  })

  test('extractDocsisStatus', () => {
    const fixtureDocsisData = fs.readFileSync(
      path.join(__dirname, './__fixtures__/status_docsis_data.php.html'),
      'utf8'
    )
    expect(
      extractDocsisStatus(
        fixtureDocsisData,
        new Date('2021-02-26T09:19:56.042Z')
      )
    ).toMatchSnapshot()
  })
  test('extractCredentialString', () => {
    const fixture = fs.readFileSync(
      path.join(__dirname, './__fixtures__/base_95x.js'),
      'utf8'
    )
    expect(extractCredentialString(fixture)).toEqual(
      'someRandomCatchyHash37f1f79255b66b5c02348e3dc6ff5fcd559654e2'
    )
  })

  test('extractFirmwareVersion', () => {
    expect(extractFirmwareVersion(fixtureIndex)).toBe('01.02.068.11.EURO.PC20')
  })
})
