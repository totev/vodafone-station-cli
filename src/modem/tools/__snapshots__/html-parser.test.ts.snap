// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmlParser extractDocsisStatus 1`] = `
{
  "downstream": [
    {
      "ChannelID": "33",
      "ChannelType": "OFDM",
      "Frequency": "151~324",
      "LockStatus": "Locked",
      "Modulation": "1024QAM",
      "PowerLevel": "-3.5/56.5",
      "SNRLevel": "40",
    },
    {
      "ChannelID": "2",
      "ChannelType": "SC-QAM",
      "Frequency": 130,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-5.2/54.8",
      "SNRLevel": 36.4,
    },
    {
      "ChannelID": "3",
      "ChannelType": "SC-QAM",
      "Frequency": 138,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-5/55",
      "SNRLevel": 36.6,
    },
    {
      "ChannelID": "4",
      "ChannelType": "SC-QAM",
      "Frequency": 146,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-5.4/54.6",
      "SNRLevel": 36.6,
    },
    {
      "ChannelID": "5",
      "ChannelType": "SC-QAM",
      "Frequency": 602,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-3/57",
      "SNRLevel": 38.6,
    },
    {
      "ChannelID": "6",
      "ChannelType": "SC-QAM",
      "Frequency": 618,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-2.8/57.2",
      "SNRLevel": 38.6,
    },
    {
      "ChannelID": "7",
      "ChannelType": "SC-QAM",
      "Frequency": 626,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-2.3/57.7",
      "SNRLevel": 39,
    },
    {
      "ChannelID": "8",
      "ChannelType": "SC-QAM",
      "Frequency": 642,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-1.7/58.3",
      "SNRLevel": 39,
    },
    {
      "ChannelID": "9",
      "ChannelType": "SC-QAM",
      "Frequency": 650,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-1.7/58.3",
      "SNRLevel": 40.4,
    },
    {
      "ChannelID": "10",
      "ChannelType": "SC-QAM",
      "Frequency": 658,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-1.5/58.5",
      "SNRLevel": 40.4,
    },
    {
      "ChannelID": "11",
      "ChannelType": "SC-QAM",
      "Frequency": 666,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-2/58",
      "SNRLevel": 39,
    },
    {
      "ChannelID": "12",
      "ChannelType": "SC-QAM",
      "Frequency": 674,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-1.2/58.8",
      "SNRLevel": 39,
    },
    {
      "ChannelID": "13",
      "ChannelType": "SC-QAM",
      "Frequency": 682,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-1.2/58.8",
      "SNRLevel": 39,
    },
    {
      "ChannelID": "14",
      "ChannelType": "SC-QAM",
      "Frequency": 690,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-0.8/59.2",
      "SNRLevel": 40.4,
    },
    {
      "ChannelID": "15",
      "ChannelType": "SC-QAM",
      "Frequency": 698,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-7.1/52.9",
      "SNRLevel": 35,
    },
    {
      "ChannelID": "16",
      "ChannelType": "SC-QAM",
      "Frequency": 706,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-6.1/53.9",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "17",
      "ChannelType": "SC-QAM",
      "Frequency": 714,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-6.1/53.9",
      "SNRLevel": 35,
    },
    {
      "ChannelID": "18",
      "ChannelType": "SC-QAM",
      "Frequency": 722,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-6.3/53.7",
      "SNRLevel": 35.5,
    },
    {
      "ChannelID": "19",
      "ChannelType": "SC-QAM",
      "Frequency": 730,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-6.5/53.5",
      "SNRLevel": 35,
    },
    {
      "ChannelID": "20",
      "ChannelType": "SC-QAM",
      "Frequency": 738,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.5/54.5",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "21",
      "ChannelType": "SC-QAM",
      "Frequency": 746,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5/55",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "22",
      "ChannelType": "SC-QAM",
      "Frequency": 754,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.7/54.3",
      "SNRLevel": 35,
    },
    {
      "ChannelID": "23",
      "ChannelType": "SC-QAM",
      "Frequency": 762,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.5/54.5",
      "SNRLevel": 34.9,
    },
    {
      "ChannelID": "24",
      "ChannelType": "SC-QAM",
      "Frequency": 770,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.6/54.4",
      "SNRLevel": 34.9,
    },
    {
      "ChannelID": "25",
      "ChannelType": "SC-QAM",
      "Frequency": 778,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.2/54.8",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "26",
      "ChannelType": "SC-QAM",
      "Frequency": 786,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.1/54.9",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "27",
      "ChannelType": "SC-QAM",
      "Frequency": 794,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.2/54.8",
      "SNRLevel": 35,
    },
    {
      "ChannelID": "28",
      "ChannelType": "SC-QAM",
      "Frequency": 802,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.5/54.5",
      "SNRLevel": 35.5,
    },
    {
      "ChannelID": "29",
      "ChannelType": "SC-QAM",
      "Frequency": 810,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.5/54.5",
      "SNRLevel": 35.5,
    },
    {
      "ChannelID": "30",
      "ChannelType": "SC-QAM",
      "Frequency": 818,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.3/54.7",
      "SNRLevel": 35.5,
    },
    {
      "ChannelID": "31",
      "ChannelType": "SC-QAM",
      "Frequency": 826,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-5.6/54.4",
      "SNRLevel": 35.7,
    },
    {
      "ChannelID": "32",
      "ChannelType": "SC-QAM",
      "Frequency": 834,
      "LockStatus": "Locked",
      "Modulation": "64QAM",
      "PowerLevel": "-6.5/53.5",
      "SNRLevel": 34.9,
    },
    {
      "ChannelID": "1",
      "ChannelType": "SC-QAM",
      "Frequency": 114,
      "LockStatus": "Locked",
      "Modulation": "256QAM",
      "PowerLevel": "-4.8/55.2",
      "SNRLevel": 35.6,
    },
  ],
  "downstreamChannels": 33,
  "ofdmChannels": 1,
  "time": "2021-02-26T09:19:56.042Z",
  "upstream": [
    {
      "ChannelID": "4",
      "ChannelType": "SC-QAM",
      "Frequency": 36,
      "LockStatus": "ACTIVE",
      "Modulation": "64QAM",
      "PowerLevel": "49/109",
    },
    {
      "ChannelID": "1",
      "ChannelType": "SC-QAM",
      "Frequency": 59,
      "LockStatus": "ACTIVE",
      "Modulation": "32QAM",
      "PowerLevel": "47/107",
    },
    {
      "ChannelID": "3",
      "ChannelType": "SC-QAM",
      "Frequency": 46,
      "LockStatus": "ACTIVE",
      "Modulation": "64QAM",
      "PowerLevel": "49/109",
    },
    {
      "ChannelID": "2",
      "ChannelType": "SC-QAM",
      "Frequency": 52,
      "LockStatus": "ACTIVE",
      "Modulation": "64QAM",
      "PowerLevel": "47/107",
    },
  ],
  "upstreamChannels": 4,
}
`;
